lib/web_ui/lib/src/engine/platform_dispatcher.dart

```diff
@@ -42,23 +84,41 @@ typedef _KeyDataResponseCallback = void Function(bool handled);
 class EnginePlatformDispatcher extends ui.PlatformDispatcher {
   /// Private constructor, since only dart:ui is supposed to create one of
   /// these.
-  EnginePlatformDispatcher._() {
+  EnginePlatformDispatcher() {
     _addBrightnessMediaQueryListener();
+    HighContrastSupport.instance.addListener(_updateHighContrast);
     _addFontSizeObserver();
+    registerHotRestartListener(dispose);
   }
 
   /// The [EnginePlatformDispatcher] singleton.
   static EnginePlatformDispatcher get instance => _instance;
-  static final EnginePlatformDispatcher _instance =
-      EnginePlatformDispatcher._();
+  static final EnginePlatformDispatcher _instance = EnginePlatformDispatcher();
 
   /// The current platform configuration.
   @override
   ui.PlatformConfiguration configuration = ui.PlatformConfiguration(
     locales: parseBrowserLanguages(),
     textScaleFactor: findBrowserTextScaleFactor(),
+    accessibilityFeatures: initializeAccessibilityFeatures(),
   );
 
+  /// Initialize accessibility features based on the current value of high contrast flag
+  static EngineAccessibilityFeatures initializeAccessibilityFeatures() {
+    final EngineAccessibilityFeaturesBuilder builder =
+        EngineAccessibilityFeaturesBuilder(0);
+    if (HighContrastSupport.instance.isHighContrastEnabled()) {
+      builder.highContrast = true;
+    }
```

Comment:
Now that we have this method, I'm wondering if we still need the logic in `set semanticsEnabled` that flips the value of `accessibleNavigation`. If we add `accessibleNavigation` here (and, in the future, all other properties), then `set semanticsEnabled` could just call this method and refresh all properties.

Reply:
This is a very good point. There are a few details I need to check before making this change. I'll keep it in mind for the next a11y PR.